localstack/services/stepfunctions/asl/parse/asl_parser.py

```diff
@@ -48,11 +48,11 @@ def __repr__(self):
 
 class AmazonStateLanguageParser(abc.ABC):
     @staticmethod
-    def parse(src: str) -> Program:
+    def parse(derivation: str) -> tuple[EvalComponent, ParserRuleContext]:
```

Comment:
I guess changing this signature is required to get more flexibility for integrating static analysis

Reply:
Correct, I will change the "derivation" naming here too